### ISSUE TITLE
Stars between ~1.5 and 6 solar masses leave massless remnants behind

### ISSUE THREAD

**Issue by ilyamandel** [TeamCOMPAS/COMPAS#865]
Bug spotted by @solfreludio : solar-Z, no-mass-loss COMPAS stars with masses between ~1.5 and ~6 Msun at ZAMS leave massless remnants behind in both SSE and BSE modes (jumping straight from type 6 -- TPAGB -- to type 15).

**Comment by jeffriley** [comment]
I believe this is what's happening:

snMass < OPTIONS->MCBUR1() in GiantBranch::ResolveSupernova(), so ResolveTypeIIaSN() is called, and that always results in a MR

The checks in GiantBranch::ResolveSupernova() are order dependent - are they in the right order?

**Comment by ilyamandel** [comment]
OK, this issue only appears when use-mass-loss = False.  In the SSE prescription that we use (see section 5.4 of Hurley, Pols, Tout, 2000), the TPAGB phase can only end in either complete envelope removal or a SN.  If we disallow envelope removal by stopping mass loss, the only choice is a SN.  Except that it’s a 2-solar-mass star, so a SN is a bit crazy, hence the "fake" IIa SN as @jeffriley pointed out.  I am not sure what the desired behaviour is under such circumstances…

**Comment by reinhold-willcox** [comment]
If this only occurs when there is no mass loss, it seems reasonable to me to use a sort of "nonsensical" SN. It's not quite an error, but it won't accidentally be included in more physically meaningful outcomes.

**Comment by jeffriley** [comment]
We could evolve things like this to a new "invalid" stellar type that would
signal users to ignore the result.

**Comment by ilyamandel** [comment]
So I am not sure whether this is an issue to be dealt with or not.  In a sense, our code is doing the right thing when forced to deal with an unphysical scenario -- so I'd say it's not a bug.  My suggestion would be to add a disclaimer that --use-mass-loss False may yield unphysical behaviour, and consider providing additional wind tuning beyond what's already available via the following, if needed:  --cool-wind-mass-loss-multiplier, --luminous-blue-variable-prescription, --luminous-blue-variable-multiplier, --mass-loss-prescription, --overall-wind-mass-loss-multiplier, --wolf-rayet-multiplier .  Thoughts?

**Comment by ilyamandel** [comment]
Fixed documentation in issue #900 as described above, which closes this issue.
